src/main/java/blackjack/domain/game/ResultGame.java

```diff
@@ -0,0 +1,100 @@
+package blackjack.domain.game;
+
+import blackjack.domain.participant.Dealer;
+import blackjack.domain.participant.Participants;
+import blackjack.domain.participant.Player;
+
+import java.util.List;
+import java.util.Map;
+
+import static blackjack.domain.game.WinTieLose.*;
+
+public class ResultGame {
+
+    private static final Integer BLACKJACK_SCORE = 21;
+    private static final String NOT_FIND_RESULT_ERROR_MESSAGE = "점수에 해당하는 승무패를 구할 수 없습니다.";
+
+    private final Map<Player, WinTieLose> playersResult;
+    private final Participants participants;
+
+    public ResultGame(final Participants participants, final Map<Player, WinTieLose> playersResult) {
+        this.participants = participants;
+        this.playersResult = playersResult;
+        calculateResult();
+    }
```

Comment:
```suggestion
   public ResultGame(Map<Player, WinTieLose> playersResult, Participants participants) {
        this.playersResult = playersResult;
        this.participants = participants;
    }
```

기본적인 생성자를 만들고 정적 팩토리 메서드를 활용해보는 것은 어떨까요?
생성자에 들어가기에는 너무 큰 로직으로 보입니다-!

위와 같이 변경하면 어떤 것들이 좋아질지 로건의 생각은 어떠신가요-?

Reply:
리팩터링 진행중에 `ResultGame` 멤버 변수에 `Participants`을 가지지 않도록 수정했습니다.
저는 정적 팩토리 메서드는 정말 필요할 때에만 만들었는데, 이번에 검색해서 찾아보니까 상속을 하지 않는 클래스가 아니라면 사용해도 무방한 것 같더라구요. 이펙티브 자바에서 생성자 대신 정적 팩토리 메서드를 고려하라라고 나와있는데, 그러면 최대한 정적 팩토리 메서드를 사용하는 것이 좋은 것일까요? 코일은 어떻게 사용하는지 궁금해요.